### ISSUE TITLE
Automattic Purge needed for Logfiles

### ISSUE THREAD

**Issue by Faar400** [CONTENIDO/CONTENIDO#400]
Ever since the smallest PHP warnings and now also deprecated hints have been written to the log files, these files have been filling up extremely quickly*. I already have files with a size of half a gigabyte (0.5 GB) that the Contenido backend can no longer handle.
*some weeks!

Therefore we need an automatic deletion/purging system that is configurable in the system or a similar configuration file.
Such like "max_errorlog_size=10k".
It should not be much bigger, because purging big textfiles is an effort.

There is a log file purge already, which deletes the lines except the last 20 lines. Maybe, this is useful?
And it's urgent to do this.

**Comment by muratpurc** [comment]
Keeping the size of the log files small is a good idea, but automatic cleaning of the log files doesn't seem to be a solution for this, since then not only warnings and deprecated messages are removed, but also errors, which can be important.

Warnings can be turned off by configuring `error_reporting ` and logging of deprecated messages by setting `$cfg['debug']['log_deprecations']` configuration.

However, I would recommend fixing the causes of the entries, as they can be an indication that plugin or module codes are not properly programmed. the code might have been perfectly fine five or ten years ago, but PHP is evolving and therefore becoming strict.

Alternatively, you can define a cron job that e.g. deletes/archives the error logs once a week, or sends an email to the administrator and then deletes them.

**Comment by Faar400** [comment]
> Keeping the size of the log files small is a good idea, but automatic cleaning of the log files doesn't seem to be a solution for this, since then not only warnings and deprecated messages are removed, but also errors, which can be important.

Yes, but if you have thousands messages of the same, it wouldn't be important to keep 999 of this messages.
And by the side, you will not find this one important error message in a 200 MB text file.

> Warnings can be turned off by configuring `error_reporting ` and logging of deprecated messages by setting `$cfg['debug']['log_deprecations']` configuration.

Yes, that's the best quick solution for the moment.

> However, I would recommend fixing the causes of the entries

as we do already, but sometimes it needs more time to fix all.

> Alternatively, you can define a cron job that e.g. deletes/archives the error logs once a week, or sends an email to the administrator and then deletes them.

possible, but would be similar to this one i wanted.
I don't want it as a must, but as a possible setting in the backend.
Why are you resisting?

**Comment by Faar400** [comment]
And by the way, the "keep 20 lines" does not work if the error files are to big.
Even show the messages does not work then. It's a white page.
But a red message in the "MyContenido" space...

**Comment by muratpurc** [comment]
> And by the way, the "keep 20 lines" does not work if the error files are to big.
> Even show the messages does not work then. It's a white page.
> But a red message in the "MyContenido" space...

This can happen when the log file size exceeds the maximum value in the configuration `$cfg['system_log']['default_memory_limit']`, which is 64MB by default.

Currently the whole log file is read into memory to erase everything but the last 20 lines in it. If the log file is too large, then this option is disabled.

But you still have the option to delete the whole log file.

**Comment by muratpurc** [comment]
> Why are you resisting?

It's not about resistance, it's about adding something that may bring little or no benefit to all other CONTENIDO users.
You want the log files to be automatically purged, others may want to be informed via an email, when the log files become very large, and others may want to archive the log files periodically, etc...

Not everyone wants the log files to be automatically deleted or purged, as they may contain information about important errors that should be analyzed and fixed. Shouldn't an administrator examine the contents of the log file beforehand?

Either way, this task would have to be put into a cron job to automatically clean up the log files. If you do this check each time before writing an entry in the log file, it would cause a significant performance problem.

CONTENIDO can be extended quite easily, what's wrong with implementing a cron job that does exactly what you want?

**Addition:**
If many other CONTENIDO users want the same thing, there is nothing wrong with implementing the configuration for the automatic purge of log files. However, as I noted earlier, we have to reckon that not everyone wants the same solution.
